ledger/ledger_test.go

```diff
@@ -0,0 +1,377 @@
+			suite.Require().NoError(err)
+			if tc.expPass {
+				suite.Require().Equal(protoSignature, aminoSignature)
+			} else {
+				suite.Require().NotEqual(protoSignature, aminoSignature)
+			}
+		})
+	}
+}
+
+func (suite *LedgerTestSuite) TestGetAddressPubKeySECP256K1() {
+	privKey, err := crypto.GenerateKey()
+	suite.Require().NoError(err)
+
+	addr := crypto.PubkeyToAddress(privKey.PublicKey)
+	expAddr, err := sdk.Bech32ifyAddressBytes("evmos", common.HexToAddress(addr.String()).Bytes())
+	suite.Require().NoError(err)
+
+	testCases := []struct {
+		name     string
+		expPass  bool
+		mockFunc func()
+	}{
+		{
+			"fail - can't find Ledger device",
+			false,
+			func() {
+				suite.ledger.PrimaryWallet = nil
+			},
+		},
+		{
+			"fail - unable to derive Ledger address",
+			false,
+			func() {
+				RegisterOpen(suite.mockWallet)
+				RegisterDeriveError(suite.mockWallet)
+			},
+		},
+		{
+			"fail - bech32 prefix empty",
+			false,
+			func() {
+				suite.hrp = ""
+				RegisterOpen(suite.mockWallet)
+				RegisterDerive(suite.mockWallet, addr, &privKey.PublicKey)
+			},
+		},
+		{
+			"pass - get ledger address",
+			true,
+			func() {
+				RegisterOpen(suite.mockWallet)
+				RegisterDerive(suite.mockWallet, addr, &privKey.PublicKey)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+			tc.mockFunc()
+			_, addr, err := suite.ledger.GetAddressPubKeySECP256K1(gethaccounts.DefaultBaseDerivationPath, suite.hrp)
+			if tc.expPass {
+				suite.Require().NoError(err, "Could not get wallet address")
+				suite.Require().Equal(expAddr, addr)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *LedgerTestSuite) TestGetPublicKeySECP256K1() {
+	privKey, err := crypto.GenerateKey()
+	suite.Require().NoError(err)
+	addr := crypto.PubkeyToAddress(privKey.PublicKey)
+	expPubkeyBz := crypto.FromECDSAPub(&privKey.PublicKey)
+	testCases := []struct {
+		name     string
+		expPass  bool
+		mockFunc func()
+	}{
+		{
+			"fail - can't find Ledger device",
+			false,
+			func() {
+				suite.ledger.PrimaryWallet = nil
+			},
+		},
+		{
+			"fail - unable to derive Ledger address",
+			false,
+			func() {
+				RegisterOpen(suite.mockWallet)
+				RegisterDeriveError(suite.mockWallet)
+			},
+		},
+		{
+			"pass - get ledger public key",
+			true,
+			func() {
+				RegisterOpen(suite.mockWallet)
+				RegisterDerive(suite.mockWallet, addr, &privKey.PublicKey)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+			tc.mockFunc()
+			pubKeyBz, err := suite.ledger.GetPublicKeySECP256K1(gethaccounts.DefaultBaseDerivationPath)
+			if tc.expPass {
+				suite.Require().NoError(err, "Could not get wallet address")
+				suite.Require().Equal(expPubkeyBz, pubKeyBz)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func verifyTypedDataFields(t *testing.T, typedData apitypes.TypedData) {
```

Comment:
@4rgon4ut `verifyTypedDataFields` is currently unused (not run in tests). Can this be removed or should the test be run?

Reply:
@MalteHerrmann as the scope of the integration tests that I working on lies on the evmos repo side, all unused code from here can be removed